### PR TITLE
Add AWS contribution Guide

### PR DESCRIPTION
**Pull request by al3pht**
# What's In This Pull Request
This pull request includes the addition of the AWS contribution guide.

# Type of Change
- [x] Enhancement (non-breaking change which improves existing functionality)
- [x] New feature (non-breaking change which adds functionality)
# Checklist
- [x] Added developer guide documentation for contributing to AWS
- [x] Tested documentation locally using `tox -e docs`
- [x] Verified the python docstring is being added to the documentation by the assistance of `.. autoclass:: c7n.query.TypeInfo`

### CODE REVIEW

File: c7n/query.py [hunk 1]
```diff
@@ -751,6 +751,8 @@ def __repr__(cls):
             identifier = cls.config_type
         elif cls.cfn_type:
             identifier = cls.cfn_type
+        elif cls.__doc__:
```
Comment (kapilt) [inline c7n/query.py:754]: what's this one for?

**Review by kapilt — COMMENTED** [review]
thanks, approach looks good

File: c7n/query.py (hunk 1 above)
Comment (al3pht) [reply c7n/query.py:754]: without it, the logic was going to the default `identifier = "AWS::%s::%s" % (cls.service.title(), cls.id)` for `class TypeInfo`, but `class TypeInfo` has no `service`, this was the only workaround I found

what do you advise?

Comment (kapilt) [reply c7n/query.py:754]: gotcha, just trying to understand the logic

**Comment by kapilt** [comment]
looks like its failing on lint checks, re `make lint` to reproduce locally.

**Comment by al3pht** [comment]
@kapilt lint issues fixed

File: c7n/query.py (hunk 1 above)
Comment (kapilt) [reply c7n/query.py]: fwiw, went ahead and updated this handling to not interrupt existing subclass identification.

**Review by kapilt — APPROVED** [review]
lgtm, thanks

**Pull request merged**